src/Utils.js

```diff
@@ -199,15 +199,15 @@ exports.clone = function (src) {
 };
 
 exports.cloneDeep = function (src) {
-    var visited = [], cloned = [];
+    var vidx = 0, visited = new Map(), cloned = [];
     function cloneDeep(src) {
         if (typeof src !== "object" || src === null) { return src; }
         var res, idx, cidx;
 
-        cidx = visited.indexOf(src);
-        if (cidx !== -1) { return cloned[cidx]; }
+        cidx = visited.get(src);
+        if (cidx !== undefined) { return cloned[cidx]; }
 
-        visited.push(src);
+        visited.set(src, vidx++);
```

Comment:
@KEIII - Perf tests would definitely help to showcase if that is indeed the case
@zaggino - Using a Map instead of an Array would certainly improve the performance, since search would happen in a constant time.